### PR TITLE
Exposed package deploy as github action

### PR DESCRIPTION
**Pull request by Dee-pak**

### CODE REVIEW

File: .github/workflows/rolling-instance-actions.yml [hunk 1]
```diff
@@ -121,6 +121,17 @@ jobs:
...
         tenant-id: ${{ env.WF_TENANT_ID }}
 
+    - name: Test deploy-package action with username-password
+      uses: ./deploy-package
+      with:
+        environment-url: ${{ steps.create-source-environment.outputs.environment-url }}
+        user-name: ${{ env.WF_USERNAME }}
+        password-secret: ${{ secrets.PASSWORD_PPDEVTOOLS }}
+        tenant-id: ${{ env.WF_TENANT_ID }}
+        package: ''
```
Comment (Dee-pak) [inline .github/workflows/rolling-instance-actions.yml:131]: I am not entirely sure what should be the path to the package here for rolling-instance-actions.

**Comment by davidjenni** [comment]
once the pac CLI update in #82 has merged, refresh this branch/PR. I hope it will reduce the "files changed" back to a non-noisy number again. :-) If not, pls remove/reset the last commit (that also refreshed the CLI) to this branch

File: deploy-package/action.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+name: 'package-deploy'
+description: 'Power Platform package deploy - Deploys the package dll or the zip file with a package.'
```
Comment (davidjenni) [inline deploy-package/action.yml:4]: add verbiage that deploy is only supported on Windows runners; the action itself should fail but nice to tell this important detail upfront :-)

File: src/test/data/pkgDeployTestData/PkgFolder/Content/en-us/EndHtml/HTML/Default.htm [hunk 3]
```diff
@@ -0,0 +1,56 @@
+﻿<html xmlns="http://www.w3.org/1999/xhtml">
```
Comment (davidjenni) [inline src/test/data/pkgDeployTestData/PkgFolder/Content/en-us/EndHtml/HTML/Default.htm:1]: please remove all the Content/ parts of the package folder in TestData. It doesn't (or at least, shouldn't) participate in the actual deploy part, as those files are more commonly used for AppSource publishing and deployment (which is way beyond of what this new deploy action does)

File: deploy-package/action.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  tenant-id:
+    description: 'Tenant id if using app-id & client secret to authenticate.'
+    required: false
+
+  package:
+    description: 'Path to a package dll or zip file with a package.'
+    required: true
+
+  log-file:
```
Comment (davidjenni) [inline deploy-package/action.yml:34]: these 2 pac CLI params for logging should NOT be exposed to the action user. Instead, we should do "the right thing (TM)" which is:
we always log to console AND we set the logfile path to a known location and upload it as part of the job's artifact. There is an npm module to do so from within an action:
https://www.npmjs.com/package/@actions/artifact#Upload-an-Artifact

https://github.com/actions/toolkit/tree/main/packages/artifact

File: src/actions/deploy-package/index.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+export async function main(factory: RunnerFactory): Promise<void> {
+    try {
+        core.startGroup('deploy-package:');
+        const platform = os.platform();
+        if (platform !== 'win32') {
+            throw Error(`Unsupported runner os: '${platform}'; package deployer is only available on Windows`);
+        }
+        const pac = factory.getRunner('pac', process.cwd());
+        const packagePath = core.getInput('package', { required: true });
+        const logToConsole = core.getInput('log-console', { required: false });
```
Comment (davidjenni) [inline src/actions/deploy-package/index.ts:22]: see my comments in action.yaml re: logConsole and logFile

**Review by davidjenni — CHANGES_REQUESTED** [review]

**Review by davidjenni — APPROVED** [review]

**Pull request merged**